atlp-rwanda/hackers-ec-be
# #187419056 Add Google authentication to enhance user login

#### What does this PR do?
This PR aims to enhance user login by adding Google authentication. By implementing this feature, users will be able to log in using their pre-existing Google accounts.

#### Description of Task to be completed?
The following endpoints need to be functional:

GET /api/v1/users/auth/google
GET /api/v1/users/auth/google/callback

#### How should this be manually tested?
To verify the changes introduced by this PR:

1. Clone the repository and navigate to the project directory.
2. Run npm install to install the necessary dependencies.
3. Check the env-example file and create a .env file, filling in the environment variables as specified in the example.
4. Start the application by running npm run dev.

5. There are two ways to test the Google authentication:

     - Use the Swagger documentation by accessing the endpoint api/v1/docs.
     - Utilize the Google authentication endpoint directly: api/v1/users/auth/google.

#### What are the relevant pivotal tracker/Trello stories?
#187419056.
